### PR TITLE
Update 4 notebooks to use the new Python SDK

### PR DESCRIPTION
**Pull request by jhills20**
## Summary

Briefly describe the changes and the goal of this PR. Make sure the PR title summarizes the changes effectively.

Updated 4 recipes to match new SDK spec

## Motivation

Why are these changes necessary? How do they improve the cookbook?
So people can run these notebooks!
---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [ ] I have added a new entry in [registry.yaml](/registry.yaml) (and, optionally, in [authors.yaml](/authors.yaml)) so that my content renders on the cookbook website.
- [ ] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [ ] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [ ] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [ ] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [ ] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [ ] Correctness: The information I include is correct and all of my code executes successfully.
  - [ ] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: examples/Clustering.ipynb [hunk 1]
```diff
@@ -21,7 +21,7 @@
        "(1000, 1536)"
       ]
      },
-     "execution_count": 1,
```
Comment (simonpfish) [inline examples/Clustering.ipynb:24]: Let's re-run from scratch and reset all of these execution counts, but asides from that, all good!

**Review by simonpfish — COMMENTED** [review]
Thank you for all of these updates!

File: examples/Embedding_long_inputs.ipynb [hunk 2]
```diff
@@ -112,23 +126,20 @@
...
...0757303238, 0.010416848585009575, -0.013582820072770119, 0.034154921770095825, -0.03898436948657036, 0.0802224799990654, 0.014863965101540089, -0.012858402915298939, 0.008344211615622044, -0.011865682899951935, 0.0015419016126543283, 0.0021531283855438232, 0.018553126603364944, -0.040191732347011566, -0.032491445541381836, -0.014796889387071133, -0.006429201457649469, 0.018110426142811775, -0.024724086746573448, -0.029566947370767593, -0.009893658570945263, 0.010369895957410336, 0.014300528913736343, -0.00897472258657217, -0.03512081131339073, 0.03361831605434418, 0.01972024142742157, 0.018754351884126663, 0.0003123210044577718, -0.010745519772171974, -0.024804577231407166, 0.01859337091445923, -0.004108383785933256, -0.021423963829874992, -0.03074747882783413, -0.014300528913736343, 0.006077054422348738, -0.00969243235886097, -0.03241095691919327, 0.0012014926178380847, -0.026897337287664413, -0.006295050028711557, -0.01153701264411211, 0.01962633617222309, 0.027581509202718735, -0.015320079401135445, 0.03965512663125992, -0.03608670085668564, 0.004866338800638914, 0.027366867288947105, 0.00013949639105703682, -0.033591486513614655, -0.003806543303653598, -0.035254962742328644]\n"
```
Comment (simonpfish) [inline examples/Embedding_long_inputs.ipynb:136]: Can we hide this output? Or only print the beginning of it?

**Review by simonpfish — CHANGES_REQUESTED** [review]

File: examples/Clustering.ipynb [hunk 3]
```diff
@@ -116,14 +118,12 @@
...
...ehQ4dmNarau3cvfu/3fg9f//rX4ff7Z0XQq4/9fv+yjnstWOhcqYyOjuL1r389brvttllBstf7XC2U2tpa6PX6inNxI83DfHzoQx/Cf/zHf+CXv/wlNmzYML3d7/cjl8shEonMsBLcqHN3+PBhjI+PY/fu3dPbisUifvnLX+J//+//jWeffVbM1xUaGxuxZcuWGdtuuukm/OAHPwCgfQeNjY2hsbFx+pixsTHs3Llz1ca5rlnrIJkbkcHBQeXkyZPTP88++6wCQPn+97+vDA0NKYqiBYjmcrnp5z3++OM3ZIDo8PCwsnHjRuVd73qXUigUZu0Xc6Wxb98+5UMf+tD042KxqDQ3N9/wwbSyLCuPPfaY0tTUpJw/f37WfjU49Pvf//70tnPnzt2QwaGKoiixWGzGd9TJkyeVvXv3Kg8//LBy8uRJMV8l/O7v/u6sYNoPf/jDyq233qooihZM+4UvfGF6fzQaFcG0i0AIlSrg0qVLs7J+IpGI0tDQoLznPe9RTp06pXznO99RbDab8tWvfnXtBroGDA8PK93d3cob3vAGZXh4WAkEAtM/KmKuNL7zne8oZrNZ+dd//VflzJkzyvvf/37F7XYrwWBwrYe2pnzwgx9UampqlF/84hczPkOpVGr6mA984ANKa2ur8vzzzyuvvvqqcuutt04vNgJlRtaPooj5Uvntb3+rGAwG5W/+5m+UCxcuKN/61rcUm82mfPOb35w+5rOf/azidruV//t//69y4sQJ5Xd+53eUjo4OJZ1Or+HI1w9CqFQBlYSKoijK8ePHlTvuuEMxm81Kc3Oz8tnPfnZtBriGPP300wqAij+liLnS+NKXvqS0trYqJpNJ2bdvn/LSSy+t9ZDWnLk+Q08//fT0Mel0Wvlv/+2/KR6PR7HZbMpb3/rWGYL4RqdcqIj50vjxj3+sbN26VTGbzcrmzZuVr33tazP2y7Ks/MVf/IXS0NCgmM1m5Q1veIPS29u7RqNdf0iKoiir7m8SCAQCgUAgWAAi60cgEAgEAkHVIoSKQCAQCASCqkUIFYFAIBAIBFWLECoCgUAgEAiqFiFUBAKBQCAQVC1CqAgEAoFAIKhahFARCAQCgUBQtQihIhAIBAKBoGoRQkUgEAgEAkHVIoSKQCAQCASCqkUIFYFAIBAIBFXL/w+sPPs/fBzEswAAAABJRU5ErkJggg==",
```
Comment (ibigio) [inline examples/Clustering.ipynb:121]: Add this in the images dir instead of inline.

File: examples/Clustering.ipynb [hunk 4]
```diff
@@ -251,6 +249,11 @@
    "source": [
     "It's important to note that clusters will not necessarily match what you intend to use them for. A larger amount of clusters will focus on more specific patterns, whereas a small number of clusters will usually focus on largest discrepencies in the data."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
```
Comment (ibigio) [inline examples/Clustering.ipynb:256]: delete empty cell

File: examples/Embedding_Wikipedia_articles_for_search.ipynb [hunk 5]
```diff
@@ -40,7 +40,11 @@
     "import openai  # for generating embeddings\n",
```
Comment (ibigio) [inline examples/Embedding_Wikipedia_articles_for_search.ipynb:40]: remove old import

File: examples/Embedding_Wikipedia_articles_for_search.ipynb [hunk 6]
```diff
@@ -40,7 +40,11 @@
...
     "import pandas as pd  # for DataFrames to store article sections and embeddings\n",
     "import re  # for cutting <ref> links out of Wikipedia articles\n",
-    "import tiktoken  # for counting tokens\n"
+    "import tiktoken  # for counting tokens\n",
+    "from openai import OpenAI\n",
+    "import openai\n",
```
Comment (ibigio) [inline examples/Embedding_Wikipedia_articles_for_search.ipynb:45]: do we need both?

File: examples/Embedding_Wikipedia_articles_for_search.ipynb [hunk 7]
```diff
@@ -645,6 +637,13 @@
     "\n",
     "df.to_csv(SAVE_PATH, index=False)\n"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
```
Comment (ibigio) [inline examples/Embedding_Wikipedia_articles_for_search.ipynb:646]: delete empty cell

File: examples/Embedding_long_inputs.ipynb [hunk 8]
```diff
@@ -28,21 +26,37 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "from openai import OpenAI\n",
     "import openai\n",
```
Comment (ibigio) [inline examples/Embedding_long_inputs.ipynb:30]: ditto: both?

File: examples/Embedding_long_inputs.ipynb [hunk 9]
```diff
@@ -249,6 +263,11 @@
    "source": [
     "In some cases, it may make sense to split chunks on paragraph boundaries or sentence boundaries to help preserve the meaning of the text."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
```
Comment (ibigio) [inline examples/Embedding_long_inputs.ipynb:270]: delete empty cell

**Review by ibigio — CHANGES_REQUESTED** [review]

File: examples/Clustering.ipynb (hunk 3 above)
Comment (jhills20) [reply examples/Clustering.ipynb:121]: @ibigio i didn't make this notebook originally but it seems like this line is from the matplotlib, not directly inline

**Review by ibigio — CHANGES_REQUESTED** [review]
Thanks for bringing this up to date! Just added a comment to move the image into the images folder to keep the code cleaner, but other than that looks good!

**Comment by sarbazvatanatan** [comment]
> ## Summary
> Briefly describe the changes and the goal of this PR. Make sure the PR title summarizes the changes effectively.
> 
> Updated 4 recipes to match new SDK spec
> 
> ## Motivation
> ## Why are these changes necessary? How do they improve the cookbook?
> So people can run these notebooks!
> ## For new content
> When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:
> 
> * [ ]  I have added a new entry in [registry.yaml](/registry.yaml) (and, optionally, in [authors.yaml](/authors.yaml)) so that my content renders on the cookbook website.
> * [ ]  I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
>   
>   * [ ]  Relevance: This content is related to building with OpenAI technologies and is useful to others.
>   * [ ]  Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
>   * [ ]  Spelling and Grammar: I have checked for spelling or grammatical mistakes.
>   * [ ]  Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
>   * [ ]  Correctness: The information I include is correct and all of my code executes successfully.
>   * [ ]  Completeness: I have explained everything fully, including all necessary references and citations.
> 
> We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

**Comment by Eldo420** [comment]
It looks like you're looking to update four Jupyter notebooks in the `openai/openai-cookbook` repository to use a new Python SDK. To achieve this, follow these steps:

### Step-by-Step Detailed Guide to Updating the Notebooks

1. **Fork the Repository**

   Navigate to the GitHub repository [openai/openai-cookbook](https://github.com/openai/openai-cookbook) and fork it to your GitHub account.

2. **Clone Your Fork**

   Clone the repository from your fork to your local machine:

    ```bash
    # Replace YOUR_USERNAME with your GitHub username
    git clone https://github.com/YOUR_USERNAME/openai-cookbook.git

    # Navigate into the cloned directory
    cd openai-cookbook
    ```

3. **Create a New Branch**

   Create and checkout a new branch where you'll make the updates:

    ```bash
    git checkout -b james/sdk_updates
    ```

4. **Make Your Changes**

   Open the Jupyter notebooks that need updating in your preferred Jupyter environment (e.g., JupyterLab, Jupyter Notebook, or VSCode with Jupyter support).

   Update the Python SDK imports and any SDK function calls as needed. Here's an example of what changes might look like:
   
   ```python
   # Old import statement
   import old_sdk

   # New import statement
   import new_sdk

   # Old usage example
   old_sdk.function_old_way()

   # New usage example
   new_sdk.function_new_way()
   ```

   Save each notebook after making the necessary updates.

5. **Save and Stage Your Changes**

   After updating the notebooks, save them and stage the changes for a commit:

    ```bash
    # Add the updated notebooks
    git add notebook1.ipynb notebook2.ipynb notebook3.ipynb notebook4.ipynb  # Change these to the actual filenames
    ```

6. **Commit Your Changes**

   Commit the changes with a descriptive message:

    ```bash
    git commit -m "Update 4 notebooks to use the new Python SDK"
    ```

7. **Push Your Changes**

   Push your branch to your GitHub fork:

    ```bash
    git push origin james/sdk_updates
    ```

8. **Create a Pull Request**

   - Go to your forked repository on GitHub.
   - You should see a notification prompting you to create a pull request for the branch you just pushed.
   - Click on the “Compare & pull request” button.
   - Add a descriptive title and detailed description for your pull request.

     Example:
     ```markdown
     ## Update 4 notebooks to use the new Python SDK

     This pull request updates four Jupyter notebooks to incorporate the latest Python SDK. Here's what has been changed:
     - Updated import statements for the new SDK.
     - Modified method calls to be compatible with the new SDK functions.
     - Verified that all cells in the notebooks run without errors using the new SDK.

     Notebooks updated:
     - `notebook1.ipynb`
     - `notebook2.ipynb`
     - `notebook3.ipynb`
     - `notebook4.ipynb`

     Closes #836
     ```

   - Once reviewed, click “Create pull request”.

9. **Link to Existing Issue**

   If this update addresses an existing issue (`#836` you mentioned), reference it in the pull request description with `Closes #836`. This will automatically link your PR to the issue and close it when the PR is merged.

### Additional Tips

- **Testing:** Before you commit and push your changes, make sure to run all cells in each notebook to verify that they work correctly with the new SDK.
- **Review:** Once the pull request is created, the repository maintainers will review it. Be prepared to make any requested changes.
- **Upstream Changes:** If other changes are made to the main repository while your PR is open, you might need to update your branch:

    ```bash
    # Add the original repository as upstream
    git remote add upstream https://github.com/openai/openai-cookbook.git

    # Fetch the upstream changes
    git fetch upstream

    # Merge the latest changes into your branch
    git merge upstream/main

    # Resolve any conflicts, then push the updated branch
    git push origin james/sdk_updates
    ```

Following these steps will help you to effectively contribute by updating the notebooks in the `openai/openai-cookbook` repository to use the new Python SDK. Good luck!

**Pull request closed without merging**